src/ts/content_scripts/injector.ts

```diff
@@ -10,6 +10,27 @@ for (const eventName of ['visibilitychange', 'webkitvisibilitychange', 'blur'])
 }
 
 async function loaded(): Promise<void> {
+  try {
+    let hostname = '';
+    if (document.referrer) {
+      const url = new URL(document.referrer);
+      hostname = url.hostname;
+    }
+
+    await holodexEnabled.loaded;
+    if (hostname === 'holodex.net' && !holodexEnabled.get()) { return; }
+  } catch (e) {
+    if (e instanceof Error) {
+      holodexEnabled.subscribe($holodexEnabled => {
+        console.log( document.referrer );
+        console.log({ $holodexEnabled });
+      });
+      console.debug('Could not get hostname', { e });
+    } else {
+      throw e;
```

Comment:
Can you remove the `throw e`? I would not want the function to crash if something goes wrong.

Also, can you replace the debugging of checking `holodexEnabled` with that singular `console.debug`?